src/Microsoft.AspNetCore.Authentication.Cookies/Events/CookieValidatePrincipalContext.cs

```diff
@@ -19,55 +19,40 @@ public class CookieValidatePrincipalContext : BaseCookieContext
         /// <param name="scheme"></param>
         /// <param name="ticket">Contains the initial values for identity and extra data</param>
         /// <param name="options"></param>
-        public CookieValidatePrincipalContext(HttpContext context, AuthenticationScheme scheme, AuthenticationTicket ticket, CookieAuthenticationOptions options)
-            : base(context, scheme, options, ticket?.Properties)
+        public CookieValidatePrincipalContext(HttpContext context, AuthenticationScheme scheme, CookieAuthenticationOptions options, AuthenticationTicket ticket)
+            : base(context, scheme, options)
         {
-            if (context == null)
-            {
-                throw new ArgumentNullException(nameof(context));
-            }
-
             if (ticket == null)
             {
                 throw new ArgumentNullException(nameof(ticket));
             }
 
-            if (options == null)
-            {
-                throw new ArgumentNullException(nameof(options));
-            }
-
             Principal = ticket.Principal;
```

Comment:
So this event is named "ValidatePrincipal", not ValidateTicket, so today its only giving the event the opportunity to manipulate the principal in the ticket, while we can certainly switch this into something like a ValidateTicket, that's not something I'm trying to do in this PR.  So we don't really need the original ticket, we return a new ticket from context.Principal and context.Properties.